From 0d4dd3688bd807c2d00f39b9db22fe904519e852 Mon Sep 17 00:00:00 2001
From: Sanyog Kale <skale@habana.ai>
Date: Sun, 05 Jun 2022 16:02:38 +0300
Subject: [PATCH] Revert "[SW-77251] gaudi3: add onix device specific mem region api"

Revert rom recovery implementation for secured firmware update via
inband and OOB.

This reverts commit bba03a463b5545aea9bc7b454952ad1e7acb1fca.

Change-Id: I5240050b1bd4273f2dcdbc5711892bc1dd6e9b73
---

diff --git a/arch/gaudi3/arc/onix.c b/arch/gaudi3/arc/onix.c
index 24e9515..7c16e57 100644
--- a/arch/gaudi3/arc/onix.c
+++ b/arch/gaudi3/arc/onix.c
@@ -5,7 +5,6 @@
  */
 
 #include <arcboot.h>
-#include <memory.h>
 
 #define SYS_CLOCK_FREQUENCY	35000000ULL
 
@@ -14,49 +13,8 @@
 	return SYS_CLOCK_FREQUENCY;
 }
 
-/*
- * Because of ONIX RAM size and layout limitation,
- * both MEM_LKD_PCI and MEM_FWUPD_PCI, are overlapped.
- *
- * This table is used by the arcboot_rom_dev_spec_mem_region.
- * the original function is defined as a weak function
- *
- */
-static struct mem_desc mem_desc[MEM_NOENT] = {
-	[MEM_GP] = {
-		.base = MEM_GP_BASE,
-		.size = MEM_GP_SZ,
-	},
-	[MEM_LKD_PCI] = {
-		.base = MEM_LKD_PCI_BASE,
-		.size = MEM_LKD_PCI_SZ,
-	},
-	[MEM_FWUPD_PCI] = {
-		.base = MEM_LKD_PCI_BASE,
-		.size = MEM_LKD_PCI_SZ,
-	},
-	[MEM_I2C] = {
-		.base = MEM_GP_BASE,
-		.size = MEM_GP_SZ,
-	},
-};
-
-int arcboot_rom_dev_spec_mem_region(enum mem_region region,
-					struct mem_desc *desc)
-{
-	if (region >= MEM_NOENT)
-		return 1;
-
-	if (!desc)
-		return 1;
-
-	desc->base = mem_desc[region].base;
-	desc->size = mem_desc[region].size;
-
-	return 0;
-}
-
 int arcboot_rom_fuse_single_die_config(void)
 {
 	return true;
 }
+
diff --git a/common/memory.c b/common/memory.c
index be008e3..46a97ec 100644
--- a/common/memory.c
+++ b/common/memory.c
@@ -27,7 +27,7 @@
 	},
 };
 
-__weak int arcboot_rom_dev_spec_mem_region(__unused enum mem_region region,
+__weak int arcboot_get_dev_spec_mem_region(__unused enum mem_region region,
 					   __unused struct mem_desc *desc)
 {
 	return 1; /* not dev spec */
@@ -51,7 +51,7 @@
 		return 1;
 
 	/* device defined memory regions */
-	err = arcboot_rom_dev_spec_mem_region(region, desc);
+	err = arcboot_get_dev_spec_mem_region(region, desc);
 	if (!err)
 		return err;
 
